libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs

```diff
@@ -266,34 +266,43 @@ private async Task<RecognizerResult> RecognizeInternalAsync(ITurnContext turnCon
             }
 
             var utterance = turnContext.Activity?.AsMessageActivity()?.Text;
+            RecognizerResult recognizerResult = null;
+            LuisResult luisResult = null;
 
-            if (string.IsNullOrWhiteSpace(utterance))
+            if (!string.IsNullOrWhiteSpace(utterance))
```

Comment:
Please invert this 'if' statement, so that the "Positive Logic" is first, and the negative logic is second. 

This is simply as it's easier for me to read and understand positive logic, especially as things get nested. :)

Reply:
Like you, I also prefer ordering if statements in a way that reduces the number of inversions/operations. I only did it this way to match the pattern seen in the rest of the SDK. That pattern always starts with `if (!string.IsNullOr...` because the "Positive Logic" is thought to be when actual text is present. When the text is not present, the else clause handles what's considered "Negative Logic" which is a sort of exception because the expected value is missing.

I'm happy to do it your way, since it's also my way and that's what I was going to do originally before changing it according to what I thought you wanted.

Reply:
I've made the changes you requested